app/alerts.py

```diff
@@ -92,46 +92,83 @@ def build_alerts_geojson():
 # The following block is dedicated to fetching information about cameras and displaying them on the map
 
 # Fetching the positions of detection units in a given department
-def get_camera_positions(dpt_code=None):
+def build_sites_markers(dpt_code=None):
 
     # As long as the user does not click on a department, dpt_code is None and we return no device
-    if not dpt_code:
-        return None
+    # if not dpt_code:
+    #     return None
 
     # We read the csv file that locates the cameras and filter for the department of interest
     camera_positions = pd.read_csv(Path(__file__).parent.joinpath('data', 'cameras.csv'), ';')
-    camera_positions = camera_positions[camera_positions['Département'] == int(dpt_code)].copy()
+    # camera_positions = camera_positions[camera_positions['Département'] == int(dpt_code)].copy()
 
-    # We build a list of dictionaries containing the info of each camera
+    # We define the site marker icon
+    icon = {
+    "iconUrl": 'https://www.flaticon.com/premium-icon/icons/svg/3371/3371105.svg',
+    "iconSize": [40, 40],  # size of the icon
+    "iconAnchor": [0, 20]  # point of the icon which will correspond to marker's location
+    # "popupAnchor": [-3, -76]  # point from which the popup should open relative to the iconAnchor
+    }
+
+    # We build a list of markers containing the info of each site/camera
     markers = []
-    for _, row in camera_positions.iterrows():
+    for i, row in camera_positions.iterrows():
         lat = row['Latitude']
         lon = row['Longitude']
-        area = row['Tours']
+        site_name = row['Tours']
         alert_codis = row['Connexion Alerte CODIS']
         nb_device = row['Nombres Devices']
-        markers.append(dict(lat=lat,
-                            lon=lon,
-                            area=area,
-                            nb_device=nb_device,
-                            alert_codis=alert_codis
-                            ))
+        markers.append(dl.Marker(id=f'site_{i}', # Necessary to set an id for each marker in order to retrieve information through callbacks
+                                 position=(lat, lon),
+                                 icon=icon,
+                                 children=[dl.Tooltip(site_name),
+                                           dl.Popup([html.H2(f'Site {site_name}'),
+                                                     html.P(f'Coordonnées : ({lat}, {lon})'),
+                                                     html.P(f'Nombre de caméras : {nb_device}')])]))
 
-    # We convert it into geojson format (not a dl.GeoJSON object yet) and return it
-    markers = dlx.dicts_to_geojson(markers)
-    markers = dlx.geojson_to_geobuf(markers)
-    return markers
+    # We group all dl.Marker objects in a dl.LayerGroup object
+    markers_layer = dl.MarkerClusterGroup(children=markers, id='sites_markers')
```

Comment:
Does this ClusterGroup creates the green clustering marker with the "2" value [here](https://user-images.githubusercontent.com/64283717/100134708-e78adf00-2e88-11eb-9560-aacc3968fb87.png) ? 

If yes did you do that on purpose for better UI when map is zoomed ?

Reply:
exactly, yes
when discussing this issue with @VincGargasson, we came to the conclusion that it became tricky to use callbacks to reproduce the behaviour: "click on a department" --> "the corresponding departments cameras appear"

At this point we imagine that the filtering on a particular department's cameras/sites will be done when the user connects to the app. Maybe we add that filter inside the `build_sites_markers' function directly as an argument.

Furthermore, when you add markers, they do not resize themselves when zooming out, so the ClusterGroup layer seemed like a nice compromise in order to not overload the map. It is completely open to debate, though!